unlock-app/src/components/interface/checkout/main/Select.tsx

```diff
@@ -439,14 +439,14 @@ export function Select({ checkoutService }: Props) {
       return
     }
 
-    const url = new URL(
-      `${window.location.protocol}//${window.location.host}${router.asPath}`
-    )
-    const params = new URLSearchParams(url.search)
-    params.set('lock', encodeURIComponent(lock.address))
-    url.search = params.toString()
-
-    router.push(url)
+    // const url = new URL(
+    //   `${window.location.protocol}//${window.location.host}${router.asPath}`
+    // )
+    // const params = new URLSearchParams(url.search)
+    // params.set('lock', encodeURIComponent(lock.address))
+    // url.search = params.toString()
+
+    // router.push(url, undefined, { shallow: true })
```

Comment:
This is what triggers a redirect. Instead of changing the URL here we should pass this as part of the redirect that happens on OAuth.